README.md

```diff
@@ -1,7 +1,37 @@
-# rds-contracts
+# RIF Marketplace Storage Pinning
 
-## message
-The `StorageProvider` can emit an arbitrary message in the function calls `setStorageOffer` and `emitMessage`. The message is is initially intended to allow the `StorageProvider` to communicate his `nodeID`, but may be used for other purposes later.
+```
+npm i @rsksmart/rif-marketplace-storage-pinning
+```
+
+## PinningManager contract
+
+### Message
+The `StorageProvider` can emit an arbitrary message in the function calls `setStorageOffer` and `emitMessage`. 
+The message is is initially intended to allow the `StorageProvider` to communicate his `nodeID`, but may be used for other purposes later.
 
 The client should be instructed to enforce the following semantics:
-- If `message` starts with `0x01`, then the following bits contain an arbitrary length `nodeID`.
\ No newline at end of file
+- If `message` starts with `0x01`, then the following bits contain an arbitrary length `nodeID`.
+
+## Glossary
+
+ - Actors
+    - **Consumer** - actor that wants his files to be persisted in the network
+    - **Provider** - actor that offers his storage for Consumers to be used in monetary exchange
+ - Entities
+    - **Offer** - Offer created by Provider that announces availability of his storage space
+      - *Billing Period* - Specifies price for recurring period of time (e.g. 10 wei per day ) 
```

Comment:
To me  **BillingPeriod** only refers to a time frame. It is not clear to me at least that the Price is included in that concept. An option could be **BillingPlan** or **SubscriptionPlan** or something like that probably. What do you think?

Reply:
Good catch. Although in this instance it is indeed meant as timeframe, because inside the contracts the pricing mechanism is tracked as period-price tuple. I think it can be confusing because developer can think that `period` means the period for how long the Agreement is supposed to be valid, so I would call it "Billing Period" instead. 

With that I would then introduce "Billing Price" as well. 

We can introduce term "Billing Plan", yet I don't think it will be used inside the smartcontracts, but if you think that it is needed for the UI let me know.

Reply:
You are right the phrasing really suggested that it is more like a Billing Plan.